### PR TITLE
Form recurring dates fixes

### PR DESCRIPTION
**Pull request by MatthewBozin**
# Description

Originally, handleChange in FormRecurringDates wasn't properly setting userData to reflect boxes being checked. This PR fixes that and creates a data structure to store all values related to recurring dates for easier consumption by the backend.

JSON example of the data structure: {'recurring': {'rate': 'weekly', 'days': ['monday', 'friday']}}

##Changes:
- Changes to handleChange method in FormRecurringDates. Only one value of weekly, monthly, or no recurring can be checked at a time. Checking a day adds the day name as a string to a days array. Unchecking a day removes the day name from the array.
- Added initial data structure for recurring dates object in userData in UserForm.js, to be used by FormRecurringDates.js.
- Added 'noRecurr' as inital value to rate property in userData.recurring. This eliminates the possibility of sending an undefined value to the backend.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by Caleb-Cohen — APPROVED** [review]
Screen Shot 2022-11-16 at 1 48 30 PM
Screen Shot 2022-11-16 at 1 48 41 PM

It does not carry the selections over to the confirmation page, but it does attempt to send the selection to the DB.

File: frontend/src/components/form/FormRecurringDates.js [hunk 1]
```diff
@@ -8,13 +8,23 @@ export default function FormRecurringDates () {
...
     const handleChange = (e) => {
-      const { name, value, type, checked } = e.target;
-        if (type === "checkbox") {
-            setUserData({ ...userData, [name]: checked })
+      const { name, value } = e.target;
+      if (value === 'rate') {
+        //ensures that only one of 'weekly', 'monthly', or 'no recurring' can be checked at a time
+        userData.recurring.rate = name
+      } else {
+        //if day is already checked, uncheck it
```
[the remark this answers is not in the document]
Comment (MatthewBozin) [reply frontend/src/components/form/FormRecurringDates.js:16]: Thanks for catching this, should be a quick fix!

[the remark this answers is not in the document]
Comment (MatthewBozin) [reply frontend/src/components/form/FormRecurringDates.js:16]: Should I go ahead and make a new PR with the added changes?

[the remark this answers is not in the document]
Comment (Caleb-Cohen) [reply frontend/src/components/form/FormRecurringDates.js:16]: I will approve.

**Pull request merged**